lib/types.d.ts

```diff
@@ -1,20 +1,32 @@
 import type {RoadmapMode} from './enums'
 
-interface IssueData {
+export interface GithubIssueData {
   body_html: string;
   body_text: string;
   body: string;
-  children: IssueData[];
-  completion_rate: number;
-  due_date: string;
-  group: string;
   html_url: string;
+  labels: string[];
   node_id: string;
-  parent: IssueData;
-  state: IssueStates;
   title: string;
+  state: IssueStates;
   root_issue?: boolean;
-  labels: string[];
+}
+
+export interface GithubIssueDataWithGroup extends GithubIssueData {
+  group: string;
+}
+
+export interface GithubIssueDataWithChildren extends GithubIssueData {
+  children: GithubIssueDataWithGroupAndChildren[] | ParserGetChildrenResponse[];
+}
+
+export interface GithubIssueDataWithGroupAndChildren extends GithubIssueDataWithGroup, GithubIssueDataWithChildren {}
+
+export interface IssueData extends GithubIssueDataWithGroupAndChildren {
+  children: IssueData[];
+  completion_rate: number;
+  due_date: string;
+  parent: IssueData;
 }
```

Comment:
omg thank you for tackling some of this. a few more iterations and I think we might be able to modify things without unexpected breakages.